src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/GameFaceDeltaObserver.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Reflection;
+using System.Threading;
+using JetBrains.Annotations;
+using UnityEngine;
+using UnityEngine.Rendering;
+using Object = UnityEngine.Object;
+
+namespace StateRecorder
+{
+    public class GameFaceDeltaObserver : MonoBehaviour
+    {
+        private const int _grayWidth = 256;
+        private const int _grayHeight = 256;
+        private byte[] _grayArray = new byte[_grayWidth * _grayHeight * 2];
+
+        private bool _firstRun = true;
+        private bool _isRecording = false;
+
+        // uses 0=false, 1=true to do interlocked threadsafe updates
+        public int _gameFaceDelta;
+
+        private RenderTexture _cothmlViewTexture;
```

Comment:
Not blocking but you know I have to point out typos...

```suggestion
        private RenderTexture _cohtmlViewTexture;
```